addons/portal/tests/test_portal.py

```diff
@@ -1,74 +1,51 @@
 # -*- coding: utf-8 -*-
-##############################################################################
-#
-#    OpenERP, Open Source Business Applications
-#    Copyright (c) 2012-TODAY OpenERP S.A. <http://openerp.com>
-#
-#    This program is free software: you can redistribute it and/or modify
-#    it under the terms of the GNU Affero General Public License as
-#    published by the Free Software Foundation, either version 3 of the
-#    License, or (at your option) any later version.
-#
-#    This program is distributed in the hope that it will be useful,
-#    but WITHOUT ANY WARRANTY; without even the implied warranty of
-#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-#    GNU Affero General Public License for more details.
-#
-#    You should have received a copy of the GNU Affero General Public License
-#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
-#
-##############################################################################
 
 from openerp.addons.mail.tests.common import TestMail
 from openerp.exceptions import AccessError
 from openerp.tools.misc import mute_logger
 
 
-class test_portal(TestMail):
+class TestPortal(TestMail):
 
     def setUp(self):
-        super(test_portal, self).setUp()
-        cr, uid = self.cr, self.uid
+        super(TestPortal, self).setUp()
+        self.MailThread = self.env['mail.thread']
+        self.ResPartner = self.env['res.partner']
 
         # Find Portal group
-        group_portal = self.registry('ir.model.data').get_object(cr, uid, 'base', 'group_portal')
-        self.group_portal_id = group_portal.id
+        self.group_portal_id = self.ref('base.group_portal')
 
         # Create Chell (portal user)
-        self.user_chell_id = self.res_users.create(cr, uid, {'name': 'Chell Gladys', 'login': 'chell', 'email': 'chell@gladys.portal', 'groups_id': [(6, 0, [self.group_portal_id])]})
-        self.user_chell = self.res_users.browse(cr, uid, self.user_chell_id)
+        self.user_chell = self.env['res.users'].with_context({'no_reset_password': True}).create({'name': 'Chell Gladys', 'login': 'chell', 'email': 'chell@gladys.portal', 'groups_id': [(6, 0, [self.group_portal_id])]})
```

Comment:
context is not updated in old test for chell user.. why with_context({'no_reset_password': True}?